Document the service binding behavior (#49740)
openshift/openshift-docs: modules/sbo-advanced-binding-options.adoc

* The `name` and `labelSelector` fields in the `binding.operators.coreos.com/v1alpha1` API.
* The `name` and `selector` fields in the `servicebinding.io` API (Spec API).
====

.Understanding the rebinding behavior

> Consider a case where after a successful binding, you are using the `name` field to identify a workload. In such a case, if you delete an existing workload and recreate it, the projected binding data are lost. The `ServiceBinding` reconciler does not rebind the workload. However, if you use the `labelSelector` field to identify a workload, the `ServiceBinding` reconciler rebinds the workload, and the Operator projects the binding data.

Reviewer: ```suggestion
Consider a case where after a successful binding, you are using the `name` field to identify a workload. In such a case, if you delete an existing workload and recreate it, the `ServiceBinding` reconciler does not rebind the workload and so the binding data are not projected to the recreated workload. However, if you use the `labelSelector` field to identify a workload, the `ServiceBinding` reconciler rebinds the workload, and the Operator projects the binding data.
```

Author: Thanks, @pmacik for the suggestion. I have implemented it but by tweaking it a little for grammar and language.

> Consider a case where after a successful binding you use the `name` field to identify a workload. In such a case, if you delete an existing workload and recreate it, the `ServiceBinding` reconciler does not rebind the workload and therefore the Operator does not project the binding data to the workload. However, if you use the `labelSelector` field to identify a workload, the `ServiceBinding` reconciler rebinds the workload, and the Operator projects the binding data.

Reviewer 2: ```suggestion
Consider a case where, after a successful binding, you use the `name` field to identify a workload. If you delete and recreate that workload, the `ServiceBinding` reconciler does not rebind the workload, and the Operator cannot project the binding data to the workload. However, if you use the `labelSelector` field to identify a workload, the `ServiceBinding` reconciler rebinds the workload, and the Operator projects the binding data.
```

Author: Done
